File: Project Academy/2. ViewModel/java/app/src/main/java/com/dicoding/academies/ui/detail/DetailCourseViewModel.java
package com.dicoding.academies.ui.detail;

import androidx.lifecycle.ViewModel;

import com.dicoding.academies.data.CourseEntity;
import com.dicoding.academies.data.ModuleEntity;
import com.dicoding.academies.utils.DataDummy;

import java.util.List;

public class DetailCourseViewModel extends ViewModel {
    private String courseId;
    private CourseEntity course;

    public void setCourseId(String courseId) {
        this.courseId = courseId;
    }

    public CourseEntity getCourse() {
        for (int i = 0; i < DataDummy.generateDummyCourses().size(); i++) {
            CourseEntity courseEntity = DataDummy.generateDummyCourses().get(i);
            if (courseEntity.getCourseId().equals(courseId)) {
                course = courseEntity;
            }
        }
        return course;
    }

    public List<ModuleEntity> getModules() {
        return DataDummy.generateDummyModules(courseId);
    }
}

File: Project Academy/2. ViewModel/kotlin/app/src/main/java/com/dicoding/academies/ui/detail/DetailCourseViewModel.kt
package com.dicoding.academies.ui.detail

import androidx.lifecycle.ViewModel

import com.dicoding.academies.data.CourseEntity
import com.dicoding.academies.data.ModuleEntity
import com.dicoding.academies.utils.DataDummy

class DetailCourseViewModel : ViewModel() {
    var courseId: String? = null
    private var course: CourseEntity? = null

    fun getCourse(): CourseEntity {
        for (i in 0 until DataDummy.generateDummyCourses().size) {
            val courseEntity = DataDummy.generateDummyCourses()[i]
            if (courseEntity.courseId == courseId) {
                course = courseEntity
            }
        }
        return course as CourseEntity
    }

    fun getModules(): List<ModuleEntity> {
        return DataDummy.generateDummyModules(courseId)
    }
}


